docs/source/usage/node_cleaning.rst

```diff
@@ -0,0 +1,259 @@
+Node Cleaning and Maintenance
+=============================
+
+Ironic uses two main methods to perform actions on a node: `in-band and out-of-band`_. Ironic supports using both methods to clean a node. Further, Ironic provides two ways of cleaning a node: automated and manual.
+
+Automated Node Cleaning
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Automated cleaning is automatically performed before the first workload has been assigned to a node and when hardware is recycled from one workload to another. Automatic cleaning can be customized for to enable or disable certain features of cleaning.
+
+Common use cases for automated cleaning:
+  * Erase disk drives and destroy all the metadata from previous user.
+  * Debug node failures resulting from disk failures by inspecting the automatic cleaning logs in Ironic conductor.
+  * Cut the time spent in manually wiping and cleaning the node after every workload, saving man hours.
+  * Reset raid configuration to a default state after every workload.
+
+Steps involved in automated cleaning:
+  1. Erase devices: out-of-band disk erase operation on all the supported physical drives in the node. Disk erase cannot be performed on physical drives.
+  2. Erase Devices Metadata: Wipe partition tables, signatures, file-system identifications, and other disk devices.
+  3. RAID Delete (Disabled-by-default): Delete any existing RAID configuration. This step needs to be enabled for automated cleaning.
+  4. Setup RAID configuration in advance(Disabled-by-default): Create RAID configuration from a predefined template when the node in enrolled. This step needs to be enabled for automated cleaning.
+
+
+Enable or Disable Automated Cleaning
+------------------------------------
+To enable automated cleaning, ensure that your ironic.conf has following option enabled.
+
+.. prompt:: bash $
+
+  [conductor]
+  automated_clean=true
+
+To disable it, ensure that your ironic.conf has following option disabled.
+
+.. prompt:: bash $
+
+  [conductor]
+  automated_clean=false
+
+Configuring Automated Cleaning Steps
+------------------------------------
+
+Automated cleaning default steps and default priorities:
+  .. list-table::
+    :widths: 25 25 50
+    :header-rows: 1
+    :align: left
+
+    * - Interface
+      - Step
+      - Priority
+    * - [deploy]
+      - erase_devices_priority
+      - 10
+    * - [deploy]
+      - erase_devices_metadata
+      - 99
+    * - [raid]
+      - delete_configuration
+      - 0
+    * - [raid]
+      - create_configuration
+      - 0
```

Comment:
If these are zero, that means that they're not run by default?

Reply:
They are not run by default, this is how default automatic cleaning configuration looks

Reply:
It just mentions the two steps but priority is set to zero, which means its disabled

Reply:
Okay - what's the difference between the two automated cleaning tables then... ?

Reply:
The first table is common to all the baremetal machines, second one has automated cleaning options available for specific hardware type.

Reply:
Okay - the "specific hardware type" is not obvious; I'd call that out in the documentation. Where would people find out what steps are supported for which hardware type?

Reply:
Added more options to the list, with a description of each item.